### PR TITLE
DataReader and DataWriter are local objects so we can just pass this …

### PR DESCRIPTION
**Pull request by jwillemsen**
…when we need a reference to a reader/writer, no need to store a seperate _var/_ptr for the reader/writer

    * dds/DCPS/DataReaderImpl.cpp:
    * dds/DCPS/DataReaderImpl.h:
    * dds/DCPS/DataReaderImpl.inl:
    * dds/DCPS/DataReaderImpl_T.h:
    * dds/DCPS/DataWriterImpl.cpp:
    * dds/DCPS/DataWriterImpl.h:
    * dds/DCPS/DataWriterImpl_T.h:
    * dds/DCPS/DiscoveryBase.h:
    * dds/DCPS/MultiTopicDataReaderBase.cpp:
    * dds/DCPS/OfferedDeadlineWatchdog.cpp:
    * dds/DCPS/OfferedDeadlineWatchdog.h:
    * dds/DCPS/PublisherImpl.cpp:
    * dds/DCPS/RecorderImpl.cpp:
    * dds/DCPS/ReplayerImpl.cpp:
    * dds/DCPS/RequestedDeadlineWatchdog.cpp:
    * dds/DCPS/RequestedDeadlineWatchdog.h:
    * dds/DCPS/SubscriberImpl.cpp:
    * dds/DCPS/SubscriberImpl.h:

### CODE REVIEW

**Comment by jwillemsen** [comment]
Changes for issue #574

**Comment by jwillemsen** [comment]
retest this please

File: dds/DCPS/DataReaderImpl.cpp [hunk 1]
```diff
@@ -224,13 +222,12 @@ DataReaderImpl::cleanup()
 }
 
 void DataReaderImpl::init(
-    TopicDescriptionImpl* a_topic_desc,
+    TopicDescriptionImpl*       a_topic_desc,
```
Comment (mitza-oci) [inline dds/DCPS/DataReaderImpl.cpp]: The coding guidelines "...Do not add extra spaces to make syntax elements (that span lines/statements) line up. This only causes unnecessary changes in adjacent lines as the code evolves."

If we're going to make changes anyway, I'd rather change it to match the guidelines.  The other option is to leave the code as-is.

Comment (jwillemsen) [inline dds/DCPS/DataReaderImpl.cpp]: Removed, will not do more methods at this moment to not trigger merge conflicts
